events/events.cwt

```diff
@@ -201,7 +208,14 @@ event = {
 	subtype[triggered] = {
 		## cardinality = 0..1
 		weight_multiplier = {
-			enum[duration] = int
+            enum[duration] = int
+
+            # TODO: Docs
+            ## cardinality = 0..1
+            bad_soldier_outcome_score = bool
+            ## cardinality = 0..1
+            title_tier_reduction_score = bool
```

Comment:
Those two sound like scripted scores.

Reply:
They're also used in some HF events so it might be a normal trigger/effect now.

Reply:
Yh, I think I'll move them.

Reply:
```	
	weight_multiplier = {
		days = 1
		bad_soldier_outcome_score = yes
		title_tier_reduction_score = yes
		modifier = {
			FROM = { trait = norse_leader }
			factor = 1.5
		}
		
		modifier = {
			factor = 2 # if the enemy leader has a particular bloodline
			FROM = {
				any_owned_bloodline = {
					has_bloodline_flag = bloodline_fearsome
				}
			}
		}
		modifier = {
			factor = 3 # if the enemy leader was a warrior lodge member
			FROM = {
				is_member_of_any_warrior_lodge_trigger = yes
				society_rank > 1 #Must be rank 2 or higher
			}
		}

		modifier = {
			factor = 0.5 # has a wonder upgrade that allows you to run away easier
			any_realm_wonder = {
				has_wonder_upgrade = upgrade_hidden_passageways
			}
		}
	}```

Reply:
Also ```	option = { #Full assault - combat_rating-based
		name = EVTOPTAHF23219
		custom_tooltip = { text = TOOLTIP_EVTOPTAHF23219 } 
		hidden_tooltip = {
			random_list = { 
				60 = {  
					good_soldier_outcome_score = yes
					character_event = { id = HF.23220 } #Success
				}
				40 = {  
					bad_soldier_outcome_score = yes
					character_event = { id = HF.23221 } #Failure
				}
			}
		} 
	}```

Reply:
I meant they're dynamically defined scores in the scripted_score_values folder and that should be covered by a generic rule already.

Reply:
Right, but I can't see them anywhere, though I'm not sure what you mean.

Reply:
Check `common/scripted_score_values/00_scripted_score_values.txt:159`; bad_soldier_outcome_score is defined there - it's a dynamic bunch of weight modifiers, similar to triggers and effects. If rules are not handling that (they're throwing errors), then check `.cwtools/common/scripted_score_values.cwt` - at the top of the file there are aliases that I've used to define them. I've split them into multiplicative and additive modifiers (because from what I understand not every weight structure can handle both types) so you just need to adjust the rules for the places you see errors at to include the corresponding modifier type that's not already covered by them.

Reply:
Right. Got it.

Reply:
Thing is it's not picking up those two then.

Reply:
Well, check the configs for the places they're used in that throw errors and see if both types of modifiers (multiplicative and additive) are allowed there. If not - add the other type, should work.

Reply:
Doesn't
```
        subtype[multiplicative] = {
            ## cardinality = 1..inf
            modifier = {

            }
        }
        subtype[additive] = {
            ## cardinality = 1..inf
            additive_modifier = {
                ## cardinality = 0..0
                localisation_key = scalar
            }
        }```
Just mean that everything without ``localisation_key = scalar`` is additive?